src/components/Form/ComponentMap.js

```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import { TextInput } from '../index'
+
+export default function GenerateComponent(componentName, props) {
```

Comment:
Ok, I'd call this a `ComponentGenerator` fwiw, but, I think for this sort of factory thing to be useful you MUST keep a library of widgets that matches your components. So it violates open/closed principle (google it) in that you'll have to keep it up to date by adding/removing components from your switch state here (I presume this is incomplete for POC purposes. But, it _would_ need to potentially support all components.

Reply:
Think about how shitty kitchensink started being to keep up to date for example-- that's the sort of high maintenance I see this becoming

Reply:
yep will push changes